packages/calcite-components/src/components/tree/tree.e2e.ts

```diff
@@ -1101,128 +1101,213 @@ describe("calcite-tree", () => {
-        });
-      }
-    );
+                    <calcite-tree-item>
+                      Grandchild 3
+                      <calcite-tree slot="children">
+                        <calcite-tree-item>Great-Grandchild 1</calcite-tree-item>
+                        <calcite-tree-item>Great-Grandchild 2</calcite-tree-item>
+                        <calcite-tree-item>Great-Grandchild 3</calcite-tree-item>
+                      </calcite-tree>
+                    </calcite-tree-item>
+                  </calcite-tree>
+                </calcite-tree-item>
+              </calcite-tree>
+            `);
+
+            const tree = await page.find("calcite-tree");
+            expect(await tree.getProperty("selectedItems")).toHaveLength(0);
+
+            const expandableParentItem = await page.find(`#${expandableItemId}`);
+            const childItems = await expandableParentItem.findAll("calcite-tree-item");
+            expect(await expandableParentItem.getProperty("expanded")).toBe(false);
+
+            await selectItem(page, expandableParentItem);
+
+            expect(await expandableParentItem.getProperty("expanded")).toBe(!selectsItem);
+            const expectedSelectedItemsAfterExpanding =
+              (selectsItem ? 1 : 0) + (selectsChildren ? childItems.length : 0);
+            expect(await tree.getProperty("selectedItems")).toHaveLength(expectedSelectedItemsAfterExpanding);
+
+            await selectItem(page, expandableParentItem);
+
+            expect(await expandableParentItem.getProperty("expanded")).toBe(false);
+            const expectedSelectedItemsAfterCollapsing = canDeselect.item ? 0 : expectedSelectedItemsAfterExpanding;
+            expect(await tree.getProperty("selectedItems")).toHaveLength(expectedSelectedItemsAfterCollapsing);
+
+            const expandableParentToggle = await page.find(`#${expandableItemId} >>> .${CSS.chevron}`);
+
+            await toggleItem(page, expandableParentItem, expandableParentToggle);
+
+            expect(await expandableParentItem.getProperty("expanded")).toBe(true);
+            expect(await tree.getProperty("selectedItems")).toHaveLength(expectedSelectedItemsAfterCollapsing);
+
+            await toggleItem(page, expandableParentItem, expandableParentToggle);
+
+            expect(await expandableParentItem.getProperty("expanded")).toBe(false);
+            expect(await tree.getProperty("selectedItems")).toHaveLength(expectedSelectedItemsAfterCollapsing);
+
+            const expandableChildItem = await page.find(`#${expandableItemChildId}`);
+
+            await selectItemChild(page, expandableChildItem);
+
+            expect(await expandableParentItem.getProperty("expanded")).toBe(
+              !selectsItem && !childToggleTraversesParent
+            );
+            expect(await tree.getProperty("selectedItems")).toHaveLength(
+              selectionMode === "none"
```

Comment:
I'm planning to clean up these assertion conditions, but might need to do in a follow-up PR.